test/server/server_test.cc

```diff
@@ -473,11 +473,24 @@ TEST_P(ServerInstanceImplTest, Stats) {
   EXPECT_EQ(2L, TestUtility::findGauge(stats_store_, "server.concurrency")->value());
   EXPECT_EQ(3L, TestUtility::findGauge(stats_store_, "server.hot_restart_epoch")->value());
 
-// This stat only works in this configuration.
-#if defined(NDEBUG) && defined(ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE)
+// The ENVOY_BUG stat only works in release mode.
+#if defined(NDEBUG)
+  ENVOY_BUG(false, "Testing envoy bug assertion failure detection in release build.");
+  EXPECT_EQ(1L, TestUtility::findCounter(stats_store_, "server.envoy_bug_failures")->value());
+  // Test power of two increments.
+  for (int i = 0; i < 16; i++) {
+    ENVOY_BUG(false);
+  }
+  EXPECT_EQ(5L, TestUtility::findCounter(stats_store_, "server.envoy_bug_failures")->value());
+
+// The ASSERT stat only works in this configuration.
+#if defined(ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE)
   ASSERT(false, "Testing debug assertion failure detection in release build.");
   EXPECT_EQ(1L, TestUtility::findCounter(stats_store_, "server.debug_assertion_failures")->value());
+#endif // defined(ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE)
```

Comment:
Possible suggestion: preserve the original structure of the server.debug_assertion_failures verification.  Having so much code under a single #ifdef NDEBUG block makes the test harder to follow.

Reply:
i agree. it's a mess. i separated the envoy bug and assert blocks.